.github/workflows/mirror-external_libs.yml

```
@@ -0,0 +1,37 @@
+# We push using git subrepo (https://github.com/ingydotnet/git-subrepo)
+# with some logic to recover from squashed parent commits
+# We first identify ourselves, needed to commit.
+# Then push to subrepo, commit to master. The commit is needed
+# to continue to replay. If we still hit issues such as this
+# action failing due to upstream changes, a manual resolution
+# PR with ./scripts/git_subrepo.sh pull will be needed.
+name: Mirror Repositories
+on:
+  schedule:
+    # Run the workflow every night at 2:00 AM UTC.
+    - cron: "0 2 * * *"
+  workflow_dispatch: {} 
+
+jobs:
+  mirror-to-noir-edwards-lib:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+          token: ${{ secrets.GITHUB_TOKEN }}
+      - name: Push to external libs repo
+        run: |
+          SUBREPO_PATH=external_libs/noir-edwards
+          git config --global user.name AztecBot
+          git config --global user.email tech@aztecprotocol.com
```

```suggestion
          git config --global user.name noirwhal
          git config --global user.email tomfrench@aztecprotocol.com
```
I'll update the email at some point